### ISSUE TITLE
Implementation details

### ISSUE THREAD

**Issue by bryanyzhu** [WaqasSultani/AnomalyDetectionCVPR2018#31]
@WaqasSultani  Hi, thank you for the awesome code base. I'm trying to train the model, but notice some minor differences between your code and the paper. 

1. Learning rate: You mentioned in your paper that you use a `lr=0.001`, but in [here](https://github.com/WaqasSultani/AnomalyDetectionCVPR2018/blob/master/TrainingAnomalyDetector_public.py#L276), you set `lr to 0.01`. 

2. Model weight L2 regularization. In your paper, you specified `lambda3=0.01`, but in [here](https://github.com/WaqasSultani/AnomalyDetectionCVPR2018/blob/master/TrainingAnomalyDetector_public.py#L31-L35), you set it to `0.001`. 

3. Did you use learning rate decay? Or you just use constant learning rate during all 20K iterations? Thanks. 

4. Is there a possibility to share your training log with me? I can get a AUC score of 74.4 really quick, like within 2K iterations, and then the accuracy remains the same. I couldn't get to 75.41 as you report in your paper. It would be really helpful if you can share your training log. 

Thank you very much, looking forward to your reply.

**Comment by mourad1081** [comment]
Also, regarding the bag instances: 

@WaqasSultani says (in the implementation details section):  "_We compute C3D features for every 16-frame video clip._" - However, their flow diagram shows that there are 32 segments videos (aka instances) per video. Does it means that the frames per instances is related to the number of frames of the video being processed. Am I wrong somewhere?

I am confused since I don't know how you exactly extract the C3D features. By looking at your samples, it seems that all files contain 32 x 4096D. Can I assume that you did something like : 

```
for each video V
    1. bag = list()
    2. nb_frames_in_one_instance = int (nb_frames in V / 32)
    3. for i in range(nb_frames_in_one_instance)
        3.1. f = generate C3D features for the frames (i*nb_frames_in_one_instance) to (i*nb_frames_in_one_instance) + nb_frames_in_one_instance.
        3.2. bag.append(f)
    4. save bag in a text file

```
?

Otherwise, I do not understand why you say "_every 16-frame video clip_".

**Comment by abdkumar** [comment]
@bryanyzhu i couldn't understand this repository code, could you share your code?
